test/common/http/conn_manager_utility_test.cc

```diff
@@ -2056,5 +2056,40 @@ TEST_F(ConnectionManagerUtilityTest, KeepFragmentFromPathWithBothOverrides) {
   EXPECT_EQ(header_map_with_fragment2.getPathValue(), "/baz/%23fragment");
 }
 
+// Verify when append_x_forward_port is turned on, the x-forwarded-port header should be appended.
+TEST_F(ConnectionManagerUtilityTest, AppendXForwardedPort) {
```

Comment:
strongly encourage an e2e test, especially if you can adapt one sending XFF

Reply:
Integration tests added.